docs/tools/mbed_cli_2/build-system/cmakelists.md

```diff
@@ -1 +1,21 @@
-# CMakelists
+# CMakeLists
+
+## Introduction
+CMake allows the description of the project by listing instructions and directives in input source files named `CMakeLists.txt`. Mbed OS has many of these files which are included in the build from a top-level `CMakeLists.txt` at the root of the project.
+We will briefly cover how `CMakeLists.txt` are used within the Mbed OS repository.
+
+## CMake target selection
+
+Mbed OS is built as a collection of all interdependent libraries essential to run the most basic application. This collection of library is used to create the CMake library target aptly named `mbed-core`. `mbed-core` is the foundation library which is linked with other libraries to make more complex systems. Applications that require an RTOS should link with the `mbed-os` CMake library target which includes `mbed-core` as well as the library that implements the RTOS. Applications destined to run on hardware with greater memory constrains and do not require an RTOS should use the `mbed-baremetal` CMake library instead of `mbed-os`.
+
+Other libraries are optional and can be linked with the application CMake target as required. To identify the name of the optional library to link with, inspect the CMake input source file (`CMakeLists.txt`) in the directory containing the source files of the library required. For example if LoRaWAN is required, inspect `connectivity/lorawan/CMakeLists.txt` and you see that the CMake library which implements it is `mbed-lorawan`.
+All Mbed CMake libraries are prefixed with `mbed-`.
+
+The CMake targets defined in the `CMakeLists.txt` of a subdirectory must be explicitly built as they are excluded from the `ALL` of the parent directory. This is because we add subdirectories using `EXCLUDE_FROM_ALL` so only the features and BSP relevant to a project and Mbed boards are excluded in the build unless needed.
+
+## Library configuration
+Source files are included in libraries by listing them in `CMakeLists.txt` files using `target_sources()`. However, header files are included by listing the directories where they can be found using `target_include_directories()`. Additionally, compile defintions can also de added to libraries using `target_compile_defintions()`
```

Comment:
`target_link_libraries()` to link targets together or add dependency?

Reply:
Yes, I can add that.

Reply:
[Done](https://github.com/ARMmbed/mbed-os-5-docs/compare/28c9170b3469fe9bbcbc353ebd68d3281401d1b8..1eee1390f710e4cf656ca3462fbc74541cb78050).